aten/src/ATen/native/cuda/GridSampler.cu

```diff
@@ -599,10 +669,20 @@ namespace {
       scalar_t iy = grid.data[grid_offset + grid_sCoor];
       scalar_t iz = grid.data[grid_offset + 2 * grid_sCoor];
 
-      // normalize ix, iy, iz from [-1, 1] to [0, inp_W-1] & [0, inp_H-1] & [0, inp_D-1]
-      float ixf = ((ix + 1.f) / 2) * (inp_W - 1);
-      float iyf = ((iy + 1.f) / 2) * (inp_H - 1);
-      float izf = ((iz + 1.f) / 2) * (inp_D - 1);
+      float ixf;
+      float iyf;
+      float izf;
+      if (align_corners) {
```

Comment:
There could be also an cuh file with similar helper functions as well. 
example: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cuda/UpSample.cuh#L140